### PR TITLE
Tc 61/topic configuration service

### PR DESCRIPTION
**Pull request by maciejmiedzianowski**

### CODE REVIEW

**Comment by MacRudDNA** [comment]
LGTM but will not approve since we worked together on it :-)

File: src/main/java/de/azapps/kafkabackup/sink/BackupSinkConfig.java [hunk 1]
```diff
@@ -1,142 +1,188 @@
...
+          ConfigDef.Importance.MEDIUM, "List of consumer groups for offset sync will be an most this old.")
+      .define(CONSUMER_OFFSET_SYNC_INTERVAL_MS, ConfigDef.Type.LONG, 60000,
+          ConfigDef.Importance.MEDIUM, "Consumer offsets will be synced at this interval.")
+      .define(MAX_BATCH_MESSAGES, ConfigDef.Type.INT, 1000,
+          ConfigDef.Importance.MEDIUM, "Batches will be rotated after this number of messages.")
+      .define(MAX_BATCH_TIME_MS, ConfigDef.Type.LONG, 600000,
+          ConfigDef.Importance.MEDIUM, "Batches will be rotated after this amount of time.")
+      .define(MIN_CONFIG_CHECK_INTERVAL_MS, ConfigDef.Type.LONG, 300000,
+          ConfigDef.Importance.MEDIUM, "Topic configuration check will be allowed after this interval.")
+      .define(TOPIC_CONFIG_BACKUP_PROPERTIES, Type.LIST, null,
```
Comment (loffek) [inline src/main/java/de/azapps/kafkabackup/sink/BackupSinkConfig.java:73]: What is the default behaviour if not specified?

File: src/main/java/de/azapps/kafkabackup/sink/BackupSinkConfig.java [hunk 2]
```diff
@@ -1,142 +1,188 @@
...
+          Importance.HIGH, "List of config properties which needs to be store in backup");
```
Comment (loffek) [inline src/main/java/de/azapps/kafkabackup/sink/BackupSinkConfig.java]: store => stored

File: src/main/java/de/azapps/kafkabackup/sink/BackupSinkConfig.java (hunk 1 above)
Comment (loffek) [reply src/main/java/de/azapps/kafkabackup/sink/BackupSinkConfig.java:73]: Ah `shouldReturnAllConfigPropertiesAsDefaultBehaviour`. Let's add that to the docstring!

File: src/main/java/de/azapps/kafkabackup/common/TopicConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+@RequiredArgsConstructor
+@Getter
+public class TopicConfiguration {
+
+  private final String topicName;
+  private final int partitionsNumber;
+  private Map<String, String> configuration;
+
+  public void setConfiguration(Map<String, String> configuration) {
+    this.configuration = new TreeMap<>(configuration);
```
Comment (loffek) [inline src/main/java/de/azapps/kafkabackup/common/TopicConfiguration.java:18]: Is the TreeMap what ensures that the config will be serialised in key-order?

**Review by loffek — CHANGES_REQUESTED** [review]
Nice! Just some very minor comments and a question :)

File: src/main/java/de/azapps/kafkabackup/common/TopicConfiguration.java (hunk 3 above)
Comment (maciejmiedzianowski) [reply src/main/java/de/azapps/kafkabackup/common/TopicConfiguration.java:18]: yes, sorted by keys

**Review by loffek — DISMISSED** [review]

**Review by loffek — APPROVED** [review]

**Pull request merged**